bson/bsoncodec/byte_slice_codec.go

```diff
@@ -23,7 +23,7 @@ type ByteSliceCodec struct {
 var (
 	defaultByteSliceCodec = NewByteSliceCodec()
 
-	_ ValueCodec  = defaultByteSliceCodec
+	_ valueCodec  = defaultByteSliceCodec
```

Comment:
This compile-time assertion is actually redundant because the codec is already passed into `RegisterTypeEncoder` and `RegisterTypeDecoder`, so I will remove it. However, I will add a comment to the `typeDecoder` assertion below.